ArjunaJTS/integration/src/main/java/com/arjuna/ats/jbossatx/BaseTransactionManagerDelegate.java

```diff
@@ -205,9 +203,10 @@ public Transaction suspend()
     // return all the event listeners associated with this thread
     private Collection<TransactionListener> getListeners(Transaction transaction, boolean create)
     {
-        if(!LISTENERS_ENABLED) {
+        if(!jtaPropertyManager.getJTAEnvironmentBean().isTransactionToThreadListenersEnabled()) {
             if(create) {
-                throw new IllegalStateException("Transaction listeners are disabled and should not be used. If you need them they can be enabled via -D" + LISTENERS_ENABLED_PROP + "=true");
+                throw new IllegalStateException(jbossatxLogger.i18NLogger.get_transaction_listeners_disabled(
```

Comment:
Can this be parameterized rather than use string concatenation?

Reply:
I understand your point but from my perspective is clearer to parametrize the message with name of property than to create API which requires some class and then a name of a property which will be later "somehow" prefixed with the class name. What if the configuration will be done differently than with the `*EnvironmentBean`s? If you really prefer doing something like 
```
@Message(id=32037,...)
public String get_transaction_listeners_disabled(Class<?> beanClass, String transactionListenersPropertyName)
```
I can change its, for sure.

Reply:
I guess as this is an exceptional circumstance it should be OK. If it was logging only we would have to avoid concatenation. My concern is purely from a performance point of view but given this is an exception it would not be normally expected